### DISCUSSION TITLE
How to setup threshold for exit-codes in Azure DevOps pipelines.

### DISCUSSION THREAD

**Discussion by clementchane** [aquasecurity/trivy#4582] [Q&A] [answered]
### Question

Hello,

I am setting up trivy in my Azure DevOps Pipeline to scan my image before publishing it to my container registry.
The scan works well but I have some critical errors that have no fixes:
![image](https://github.com/aquasecurity/trivy/assets/112370548/acf10987-1ec7-400a-bbce-bc39fb84789e)

I would like to setup an exit-code 1 to fail the pipeline if more than $thresholdcritical is found for example.
For example this would fail the pipeline if more than 10 critical vulnerabilities are found, else continue.:
`trivy image --exit-code 1 --severity critical --threshold 10 $image `

Is there any way to achieve that?
I tried to look at the doc and the github discussions but didnt find any answers.

Thanks,
Regards

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Version

```bash
0.42.0
```

**Comment by knqyf263** [comment, marked as answer]
Not supported. You can export JSON and use jq or something lie that.

**Reply by clementchane** [reply]
Ok thanks for the confirmation! 
It was what I was planning to do !

**Reply by AnaisUrlichs** [reply]
Hi @clementchane would be great if you can share the solution you came up with/that you will use with the community :)

**Reply by clementchane** [reply]
Hello,

Sure, here is what I ve written in my ADO pipeline.
Not the cleanest code as I m not so used to jq, but it s running fine :)
I have also added the part to publish the results since many blogs are out of date, this one is running properly

```
        - script: |
            trivy image --severity HIGH,CRITICAL $(containerregistry)/$(imageRepository):$(Build.SourceBranchName)-$(product_version) --format template --template "@junit.tpl" -o $(System.ArtifactsDirectory)/junit-report-crit-high-app.xml --ignore-unfixed
          displayName: "Image scan by Trivy"
          condition: and(succeeded(), eq('${{ parameters.run_trivy_vulnerability_scan }}', 'true'))
        
        - script: |
            trivy image --ignore-unfixed -s CRITICAL -f json -o critresults.json $(containerregistry):/$(imageRepository):$(Build.SourceBranchName)-$(product_version)
            jq -r '.Results' critresults.json > temp.json
            javavul=$(jq 'map(select(.Target == "Java").Vulnerabilities[]) | length' temp.json)
            pythonvul=$(jq 'map(select(.Target == "Python").Vulnerabilities[]) | length' temp.json)
            (( javavul += pythonvul))
            if (( javavul < $(critthreshold) ));
            then
              echo "Less than $(critthreshold)critical vulnerabilities";
            else
              echo "More than $(critthreshold)critical vulnerabilities";
              exit 1
            fi
          displayName: "Setting threshold on critical vulnerabilities"
          condition: and(succeeded(), eq('${{ parameters.run_trivy_vulnerability_scan }}', 'true'))

        - task: PublishTestResults@2
          displayName: "Publish trivy high-crit report"
          inputs:
            testResultsFormat: 'JUnit'
            testResultsFiles: "$(System.ArtifactsDirectory)/junit-report-crit-high-app.xml"
            mergeTestResults: true
            failTaskOnFailedTests: false
            testRunTitle: 'Trivy - High and Critical Vulnerabilities'
          condition: and(succeeded(), eq('${{ parameters.run_trivy_vulnerability_scan }}', 'true'))
```
